src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java

```diff
@@ -540,8 +540,11 @@ private int tryClickContainer(boolean isShiftKeyDown) {
                 FluidActionResult result = FluidUtil.tryEmptyContainer(currentStack, fluidTank, Integer.MAX_VALUE, null,
                         false);
                 if (!result.isSuccess()) break;
-                ItemStack remainingStack = FluidUtil
-                        .tryEmptyContainer(currentStack, fluidTank, Integer.MAX_VALUE, null, true).getResult();
+
+                result = FluidUtil.tryEmptyContainer(currentStack, fluidTank, Integer.MAX_VALUE, null, true);
+                if (!result.isSuccess()) break;
+
```

Comment:
You need to actually *do* the transfer too.

Reply:
Are you confused or am I confused?

Reply:
Oh, sorry. The last argument to `tryEmptyContainer` is `doDrain` not `simulate`. Classic forge swapping semantics.
Does this actually fix it without transferring duped fluid to the tank widget? The original simulate call should handle it no?

Reply:
It *should*[tm], but it somehow doesn't. Instead of "failing" it "succeeds with filled bucket", which causes the original .isSuccess() to return true, where it *shouldn't*. When executing, the bucket will instead actually "fail", which aborts the whole transaction.